### PR TITLE
scripts: add 'BOM' update script

### PR DESCRIPTION
**Pull request by gyuho**
Need this script when we add external dependencies.

c.f. https://github.com/coreos/etcd/pull/7896

### CODE REVIEW

File: scripts/updatebom.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+set -e
+
+if ! [[ "$0" =~ "scripts/updatebom.sh" ]]; then
+	echo "must be run from repository root"
+	exit 255
+fi
+
+echo "updating 'bill-of-materials.json'"
+go get -d -u github.com/coreos/license-bill-of-materials
+SHA_BOM="61fa1b0553b9fc2ddf8d292e03f405abdc3f6ac7"
```
Comment (heyitsanthony) [inline scripts/updatebom.sh]: probably shouldn't hardcode the SHA since this might mask BOM bugs and CI uses master

File: scripts/updatebom.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+if ! [[ "$0" =~ "scripts/updatebom.sh" ]]; then
+	echo "must be run from repository root"
+	exit 255
+fi
+
+echo "installing 'bill-of-materials.json'"
+go get -v -u github.com/coreos/license-bill-of-materials
+
+# TODO: remove 5 lines from "project": "github.com/coreos/etcd/cmd/etcd"
+license-bill-of-materials \
```
Comment (heyitsanthony) [inline scripts/updatebom.sh]:
```sh
mkdir gopath/
ln -s  gopath/src
GOPATH=`pwd`/gopath license-bill-of-materials \
    --override-file ./bill-of-materials.override.json \
    github.com/coreos/etcd github.com/coreos/etcd/etcdctl > ./bill-of-materials.json
rm -rf gopath
```
?

**Comment by fanminshi** [comment]
```
Checking bill of materials...
2a3,7
> 		"project": "bitbucket.org/ww/goautoneg",
> 		"license": "BSD 3-clause \"New\" or \"Revised\" License",
> 		"confidence": 1
> 	},
> 	{
205,209d209
< 		"confidence": 1
< 	},
< 	{
< 		"project": "bitbucket.org/ww/goautoneg",
< 		"license": "BSD 3-clause \"New\" or \"Revised\" License",
vendored licenses do not match given bill of materials
```
via https://travis-ci.org/coreos/etcd/jobs/230138438

File: scripts/updatebom.sh (hunk 2 above)
Comment (gyuho) [inline scripts/updatebom.sh]: Doesn't seem to work with symlinks of symlinks?

```bash
rm -rf ./gopath
mkdir -p ./gopath
ln -s ./cmd/vendor ./gopath/src

GOPATH=`pwd`/gopath license-bill-of-materials \
    --override-file ./bill-of-materials.override.json \
    github.com/coreos/etcd github.com/coreos/etcd/etcdctl
```

Will try other commands as well

Comment (gyuho) [inline scripts/updatebom.sh]:
```
2017/05/08 16:01:06 can't load package: package github.com/coreos/etcd: cannot find package "github.com/coreos/etcd" in any of:
	/usr/local/go/src/github.com/coreos/etcd (from $GOROOT)
	/Users/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd (from $GOPATH)
can't load package: package github.com/coreos/etcd/etcdctl: cannot find package "github.com/coreos/etcd/etcdctl" in any of:
	/usr/local/go/src/github.com/coreos/etcd/etcdctl (from $GOROOT)
	/Users/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdctl (from $GOPATH)
```

**Comment by gyuho** [comment]
PTAL.

This should fix the current Travis failures.

**Comment by heyitsanthony** [comment]
lgtm thanks

**Pull request merged**